File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string =
    let s = [ "Pling"; "Plang"; "Plong" ]
    let d = [ 3; 5; 7 ]

    Seq.zip d s
    |> Seq.map (fun (d, s) ->
        if number % d = 0 then s else "")
    |> Seq.reduce (+)
    |> fun r ->
        if r <> "" then r else string number

File: python/raindrops/raindrops.py
def convert(n):
    """Convert a number to a string according to raindrop sound values."""
    drops = ((3, 'Pling'), (5, 'Plang'), (7, 'Plong'))

    spoke = [speak for value, speak in drops if n % value == 0]
    return "".join(spoke) if spoke else str(n)
